# Kickstarter: Exploration & Prediction
By:
* Cen Liang
* Mahmoud Belhaj
* Jonas Cristens

### Load the required packages

In [18]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

### Read the Kickstarter projects file 201801 csv and parse the dates and change the categorical columns to categories

In [24]:
df = pd.read_csv('ks-projects-201801.csv', parse_dates = ['launched', 'deadline'], index_col = 'ID')
df = df.astype({"category": 'category',
                "main_category": 'category', 
                "currency": 'category', 
                'state' : 'category', 
                'country' : 'category',
               })
df.head()

name        category  \
ID                                                                              
1000002330                    The Songs of Adelaide & Abullah          Poetry   
1000003930      Greeting From Earth: ZGAC Arts Capsule For ET  Narrative Film   
1000004038                                     Where is Hank?  Narrative Film   
1000007540  ToshiCapital Rekordz Needs Help to Complete Album           Music   
1000011046  Community Film Project: The Art of Neighborhoo...    Film & Video   

           main_category currency   deadline     goal            launched  \
ID                                                                          
1000002330    Publishing      GBP 2015-10-09   1000.0 2015-08-11 12:12:28   
1000003930  Film & Video      USD 2017-11-01  30000.0 2017-09-02 04:43:57   
1000004038  Film & Video      USD 2013-02-26  45000.0 2013-01-12 00:20:50   
1000007540         Music      USD 2012-04-16   5000.0 2012-03-17 03:24:11   
1000011046  Film & Video      USD 2015-08-29  19500.0 2015-07-04 08:35:03   

            pledged     state  backers country  usd pledged  usd_pledged_real  \
ID                                                                              
1000002330      0.0    failed        0      GB          0.0               0.0   
1000003930   2421.0    failed       15      US        100.0            2421.0   
1000004038    220.0    failed        3      US        220.0             220.0   
1000007540      1.0    failed        1      US          1.0               1.0   
1000011046   1283.0  canceled       14      US       1283.0            1283.0   

            usd_goal_real  
ID                         
1000002330        1533.95  
1000003930       30000.00  
1000004038       45000.00  
1000007540        5000.00  
1000011046       19500.00

### Check if there is data missing

In [25]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 378661 entries, 1000002330 to 999988282
Data columns (total 14 columns):
name                378657 non-null object
category            378661 non-null category
main_category       378661 non-null category
currency            378661 non-null category
deadline            378661 non-null datetime64[ns]
goal                378661 non-null float64
launched            378661 non-null datetime64[ns]
pledged             378661 non-null float64
state               378661 non-null category
backers             378661 non-null int64
country             378661 non-null category
usd pledged         374864 non-null float64
usd_pledged_real    378661 non-null float64
usd_goal_real       378661 non-null float64
dtypes: category(5), datetime64[ns](2), float64(5), int64(1), object(1)
memory usage: 31.1+ MB


There is no missing data, there in total 378661 rows and 16 columns.

In [26]:
df.describe()

goal       pledged        backers   usd pledged  \
count  3.786610e+05  3.786610e+05  378661.000000  3.748640e+05   
mean   4.908079e+04  9.682979e+03     105.617476  7.036729e+03   
std    1.183391e+06  9.563601e+04     907.185035  7.863975e+04   
min    1.000000e-02  0.000000e+00       0.000000  0.000000e+00   
25%    2.000000e+03  3.000000e+01       2.000000  1.698000e+01   
50%    5.200000e+03  6.200000e+02      12.000000  3.947200e+02   
75%    1.600000e+04  4.076000e+03      56.000000  3.034090e+03   
max    1.000000e+08  2.033899e+07  219382.000000  2.033899e+07   

       usd_pledged_real  usd_goal_real  
count      3.786610e+05   3.786610e+05  
mean       9.058924e+03   4.545440e+04  
std        9.097334e+04   1.152950e+06  
min        0.000000e+00   1.000000e-02  
25%        3.100000e+01   2.000000e+03  
50%        6.243300e+02   5.500000e+03  
75%        4.050000e+03   1.550000e+04  
max        2.033899e+07   1.663614e+08

### Check the data types

In [28]:
df.dtypes

name                        object
category                  category
main_category             category
currency                  category
deadline            datetime64[ns]
goal                       float64
launched            datetime64[ns]
pledged                    float64
state                     category
backers                      int64
country                   category
usd pledged                float64
usd_pledged_real           float64
usd_goal_real              float64
dtype: object

The dtypes are correct now we can calculate the difference in days between `launced` and `deadline`.

In [29]:
df['days_to_collect'] = (df['deadline'] - df['launched']).dt.days
df['days_to_collect'].head(5)

ID
1000002330    58
1000003930    59
1000004038    44
1000007540    29
1000011046    55
Name: days_to_collect, dtype: int64

### What are the different Kickstarter states?

In [30]:
df['state'].unique()

[failed, canceled, successful, live, undefined, suspended]
Categories (6, object): [failed, canceled, successful, live, undefined, suspended]

We define 
* `failed`, `canceled`, `undefined` and `suspended` as failed.
* `successful` as success.
* `live` not relevant ==> will be deleted since our goal is to predict if it will fail or succeed